### ISSUE TITLE
Minimal APIs

### ISSUE THREAD

**Issue by andreatosato** [khellang/Middleware#157]
Hi, Can I use ProblemDetails in minimal api?
I have this error:
```txt
Unable to resolve service for type 'Microsoft.AspNetCore.Mvc.Infrastructure.IActionResultExecutor`1[Microsoft.AspNetCore.Mvc.ObjectResult]'
while attempting to activate 'Hellang.Middleware.ProblemDetails.ProblemDetailsMiddleware'
```

**Comment by marcominerva** [comment]
The issues is related to the fact the the `IActionResultExecutor<ObjectResult>` interface is registered in the `AddMvcCore `extension method (that, in turn, is automatically called by `AddControllers`): https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs#L248. As you're working with Minimal APIs, you don't have this registration, so you get this exception.

The easiest way to solve the problem is to provide an implementation of this interface that is valid for Minimal APIs:

```
/// <summary>
/// Executes an <see cref="MinimalObjectResultExecutor"/> to write to the response.
/// </summary>
public class MinimalObjectResultExecutor : IActionResultExecutor<ObjectResult>
{
    /// <summary>
    /// Executes the <see cref="MinimalObjectResultExecutor"/>.
    /// </summary>
    /// <param name="context">The <see cref="ActionContext"/> for the current request.</param>
    /// <param name="result">The <see cref="ObjectResult"/>.</param>
    /// <returns>
    /// A <see cref="Task"/> which will complete once the <see cref="ObjectResult"/> is written to the response.
    /// </returns>
    public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
    {
        ArgumentNullException.ThrowIfNull(context);
        ArgumentNullException.ThrowIfNull(result);

        var executor = Results.Json(result.Value, null, "application/problem+json", result.StatusCode);
        return executor.ExecuteAsync(context.HttpContext);
    }
}
```

Then you need to add it to the service collection, for example before calling the `AddProblemDetails` method:

```
builder.Services.TryAddSingleton<IActionResultExecutor<ObjectResult>, MinimalObjectResultExecutor>();
builder.Services.AddProblemDetails();
```

Now everything will work like a charm.

**Comment by khellang** [comment]
Nice solution @marcominerva! 👍🏻 

It's unfortunate that the middleware has a dependency on MVC infrastructure. Ideally, Project Houdini would've gotten further off the ground and pushed more of the conneg bits further down the layers by now 😞

I've thought about implicitly adding a registration for `IActionResultExecutor<ObjectResult>` in `AddProblemDetails` to counter this problem, but haven't gotten that far. Maybe the introduction of Minimal APIs will push me over the line 😅

**Comment by agilenut** [comment]
I'd support adding the registration in `AddProblemDetails`. I guess you'd want to make it such that the registration didn't mess up the actual implementation if `AddMvcCore` is called after `AddProblemDetails`.

**Comment by khellang** [comment]
This has been fixed with v6.5.0 😄

**Comment by Jerunnon** [comment]
Apparently this bug is back. I created a clean project with default WebApi Template and added Hellang Middleware Version 6.5.1 and I used the provided example to structure the program.cs.

Error:

Unhandled exception. System.AggregateException: Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Microsoft.AspNetCore.Mvc.Infrastructure.IActionResultExecutor`1[Microsoft.AspNetCore.Mvc.ObjectResult] Lifetime: Singleton ImplementationType: Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor': Unable to resolve service for type 'Microsoft.AspNetCore.Mvc.Infrastructure.OutputFormatterSelector' while attempting to activate 'Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor'.)
 ---> System.InvalidOperationException: Error while validating the service descriptor 'ServiceType: Microsoft.AspNetCore.Mvc.Infrastructure.IActionResultExecutor`1[Microsoft.AspNetCore.Mvc.ObjectResult] Lifetime: Singleton ImplementationType: Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor': Unable to resolve service for type 'Microsoft.AspNetCore.Mvc.Infrastructure.OutputFormatterSelector' while attempting to activate 'Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor'.
 ---> System.InvalidOperationException: Unable to resolve service for type 'Microsoft.AspNetCore.Mvc.Infrastructure.OutputFormatterSelector' while attempting to activate 'Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(ServiceIdentifier serviceIdentifier, Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, ServiceIdentifier serviceIdentifier, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, ServiceIdentifier serviceIdentifier, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(ServiceDescriptor serviceDescriptor, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ValidateService(ServiceDescriptor descriptor)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ValidateService(ServiceDescriptor descriptor)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider..ctor(ICollection`1 serviceDescriptors, ServiceProviderOptions options)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.DependencyInjection.ServiceProvider..ctor(ICollection`1 serviceDescriptors, ServiceProviderOptions options)
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionContainerBuilderExtensions.BuildServiceProvider(IServiceCollection services, ServiceProviderOptions options)
   at Microsoft.Extensions.Hosting.HostApplicationBuilder.Build()
   at Microsoft.AspNetCore.Builder.WebApplicationBuilder.Build()
   at Program.<Main>$(String[] args) in /Users/simonklein/Library/Mobile Documents/com~apple~CloudDocs/WebDevelopment/codetutors/MinimalApiTest/MinimalApiTest/Program.cs:line 41

Process finished with exit code 134.

Any fix for this?
